### PR TITLE
New create list (draft)

### PR DESCRIPTION
**Pull request by Joseph-Cashmore**
An attempt at getting the POST list request working. I seem to have one error left to do with a mismatch calling strings and pointers. Not sure if I know enough to get it fixed on my own.

### CODE REVIEW

File: handlers/postList.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	}
+
+	list := map[string]string{"listID": listID}
+
+	return list, http.StatusOK
+}
+
+func (g *postList) createList(path string) (string, error) {
+	listID := generateListID(path)
+	return listID, g.db.CreateList(listID)
```
Comment (samstradling) [inline handlers/postList.go:42]: CreateList takes a pointer to a string:

https://github.com/mount-joy/thelist-lambda/blob/81e77cce24a11e3ecacd1876907ff082bbc154c9/db/createList.go#L8

So this should be:

```suggestion
	return listID, g.db.CreateList(&listID)
```

**Comment by samstradling** [comment]
@Joseph-Cashmore I think something odd has happened with the branches here - maybe from branching off a PR that was squashed as it was merged. Let me know if you need a hand fixing it.

**Comment by Joseph-Cashmore** [comment]
Having some real problems with docker, so unfortunately I haven't been able to test this, but it builds successfully. I think more experienced hands are needed to get this to an acceptable level.

File: db/createList.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package db
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/dynamodb"
+)
+
+func (d *dynamoDB) CreateList(listID *string) (error) {
```
Comment (alexNaylorTeece) [inline db/createList.go:8]: I think we are no longer using a pointer to string

File: db/createList.go [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+	input := &dynamodb.PutItemInput{
+		Item: map[string]*dynamodb.AttributeValue{
+			"Id": { S: listID }, "Name": { S: aws.String("Shopping List") },
```
Comment (alexNaylorTeece) [inline db/createList.go:11]: If you change `listID` that's passed in not to be a pointer, you will need to do `&listID` here

Comment (alexNaylorTeece) [reply db/createList.go:11]: Do we want to hard code `Shopping List` or should that be passed in as `listName` or something?

File: handlers/postList.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	db db.DB
+}
+
+func newPostList() RouteHandler {
+	return &postList{
+		db: db.DynamoDB(),
+	}
+}
+
+func (g *postList) match(request events.APIGatewayProxyRequest) bool {
```
Comment (alexNaylorTeece) [inline handlers/postList.go:22]: Think this `g` would be a `p` (I'm assuming this is copid from `getList` or something

File: handlers/postList.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	return request.Path == "/lists" && request.HTTPMethod == "POST"
+}
+
+func (g *postList) handle(request events.APIGatewayProxyRequest) (interface{}, int) {
```
Comment (alexNaylorTeece) [inline handlers/postList.go:26]: ditto with the `g`

Comment (alexNaylorTeece) [reply handlers/postList.go:26]: This will mean you need to change g to p in a few places

File: handlers/postList.go [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+import (
+	"log"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/mount-joy/thelist-lambda/db"
+)
+
+type postList struct {
```
Comment (alexNaylorTeece) [inline handlers/postList.go:12]: create list or post list?

**Review by alexNaylorTeece — CHANGES_REQUESTED** [review]

**Pull request closed without merging**